### PR TITLE
update MMPD

### PR DESCRIPTION
**Pull request by McJackTang**
enable label choose

### CODE REVIEW

File: configs/infer_configs/PURE_UNSUPERVISED.yaml [hunk 1]
```diff
@@ -8,8 +8,8 @@ UNSUPERVISED:
     DATASET: PURE
     DO_PREPROCESS: False               # if first time, should be true
     DATA_FORMAT: NDHWC
-    DATA_PATH: "/gscratch/ubicomp/xliu0/data3/mnt/Datasets/PURE/RawData"                     # Raw dataset path, need to be updated
-    CACHED_PATH: "/gscratch/ubicomp/xliu0/data3/mnt/Datasets/rppg_toolbox/PreprocessedData"    # Processed dataset save path, need to be updated
+    DATA_PATH: "/data1/acsp/toolbox_data/PURE/RawData"           # Raw dataset path, need to be updated
```
Comment (xliucs) [inline configs/infer_configs/PURE_UNSUPERVISED.yaml:11]: revert them back.

File: configs/infer_configs/PURE_UNSUPERVISED.yaml [hunk 2]
```diff
@@ -8,8 +8,8 @@ UNSUPERVISED:
...
+    CACHED_PATH: "/data/rPPG_dataset/processed_dataset_makeup"   # Processed dataset save path, need to be updated
```
Comment (xliucs) [inline configs/infer_configs/PURE_UNSUPERVISED.yaml:12]: same

File: dataset/data_loader/BaseLoader.py [hunk 3]
```diff
@@ -206,7 +206,7 @@ def face_detection(self, frame, use_larger_box=False, larger_box_coef=1.0):
         """
 
         detector = cv2.CascadeClassifier(
-            './dataset/haarcascade_frontalface_default.xml')
+            '/data/rPPG-Toolbox_MMPD/dataset/haarcascade_frontalface_default.xml')
```
Comment (xliucs) [inline dataset/data_loader/BaseLoader.py:209]: Don't change it.

**Review by xliucs — CHANGES_REQUESTED** [review]
Can you please provide some sample results for at least one training / inference results? Also please consider adding pre-trained models. Thanks.

**Comment by McJackTang** [comment]
METHODS | MAE | RMSE | MAPE | PEARSON
TS-CAN(PURE_MMPD(simplest)) | 1.78 | 3.57 | 2.47 | 0.93
TS-CAN(UBFC__MMPD(simplest)) | 1.46 | 3.13 | 2.04 | 0.94

**Review by yahskapar — CHANGES_REQUESTED** [review]
Left a few comments that I think should be addressed prior to merging. Also, I'm a bit confused by the general approach with settings in the config files on second glance...for something like motion I see what's below:

```
MOTION: [1, 2, 3, 4] # 'Stationary', 'Rotation', 'Talking', 'Walking'
```
Why can't this be the following...
```
MOTION: ['Stationary', 'Rotation', 'Talking', 'Walking']
```
So that the setting is practically self-explanatory without the comment? Similarly for things like GENDER, why use 1 and 2 instead of just Male and Female? For GLASSES, why 1 and 2 instead of just True or False? I feel like the usage of numbers here, except in the case of SKIN_COLOR, doesn't really make sense.

File: config.py [hunk 4]
```diff
@@ -32,6 +32,15 @@
 _C.TRAIN.MODEL_FILE_NAME = ''
 # Train.Data settings
 _C.TRAIN.DATA = CN()
+_C.TRAIN.DATA.INFO = CN()
+_C.TRAIN.DATA.INFO.LIGHT = ['']
+_C.TRAIN.DATA.INFO.MOTION = ['']
+_C.TRAIN.DATA.INFO.EXERCISE = [True]
+_C.TRAIN.DATA.INFO.SKIN_COLOR = [1]
+_C.TRAIN.DATA.INFO.GENDER = ['']
+_C.TRAIN.DATA.INFO.GLASSER = [True]
```
[the remark this answers is not in the document]
Comment (McJackTang) [reply config.py:41]: Thanks for advice. For the label settings, we used to set as what you recommended but this is more accessible for users with less input. For typo, we define the glasser as the person who wears glasses, and this is due to our dataset labels which has already be shared.

File: configs/train_configs/UBFC_UBFC_MMPD_TSCAN_BASIC.yaml [hunk 5]
```diff
@@ -1,5 +1,5 @@
 BASE: ['']
-TOOLBOX_MODE: "train_and_test"      # "train_and_test"  or "only_test"
+TOOLBOX_MODE: "only_test"      # "train_and_test"  or "only_test"
```
[the remark this answers is not in the document]
Comment (McJackTang) [reply configs/train_configs/UBFC_UBFC_MMPD_TSCAN_BASIC.yaml:2]: thanks for reminding for the config setting

File: configs/train_configs/MMPD_MMPD_UBFC_TSCAN_BASIC.yaml [hunk 6]
```diff
@@ -2,15 +2,24 @@ BASE: ['']
 TOOLBOX_MODE: "train_and_test"      # "train_and_test"  or "only_test"
 TRAIN:
   BATCH_SIZE: 4
-  EPOCHS: 10
+  EPOCHS: 20
```
[the remark this answers is not in the document]
Comment (McJackTang) [reply configs/train_configs/MMPD_MMPD_UBFC_TSCAN_BASIC.yaml:5]: Yeah we have changed that for quicker training. I would change this back. Thanks

File: configs/train_configs/SCAMPS_SCAMPS_MMPD_TSCAN_BASIC.yaml
[the remark this answers is not in the document]
Comment (McJackTang) [reply configs/train_configs/SCAMPS_SCAMPS_MMPD_TSCAN_BASIC.yaml:1]: Sure I would check it. Thanks!

File: config.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (McJackTang) [reply config.py:41]: Furthermore, the digital numbers of labels have benefit for our file filter. Considering all aspects, I may suggest maintaining the settings of labels and glasser.

File: dataset/data_loader/MMPDLoader.py [hunk 7]
```diff
@@ -118,14 +122,116 @@ def preprocess_dataset_subprocess(self, data_dirs, config_preprocess, i, file_li
...
+            motion = 1
+        elif information[1] == 'Rotation':
+            motion = 2
+        elif information[1] == 'Talking':
+            motion = 3
+        elif information[1] == 'Walking':
+            motion = 4
+        
+        exercise = ''
+        if information[2] == 'True' or information[2] == True:
```
Comment (yahskapar) [inline dataset/data_loader/MMPDLoader.py:205]: For all of the cases here and below where we are evaluating True or False, why do we check for both the string 'True' and the Boolean True? Shouldn't we just check for one of the two?

In the case of MMPD, it looks like it's evaluating a string 'True' or 'False', so just check for that.

**Review by yahskapar — CHANGES_REQUESTED** [review]
Some small changes I think you should make and a few fixes - thank you for your work on this PR and responding to my feedback!

**Comment by McJackTang** [comment]
Thanks for all the suggestions. Really appreciate  it.
I will fix the bugs and typos according  to your advice and the convience of all users.

File: configs/train_configs/SCAMPS_SCAMPS_MMPD_TSCAN_BASIC.yaml [hunk 8]
```diff
@@ -32,19 +32,19 @@ VALID:
     FS: 30
     DATASET: SCAMPS
     DO_PREPROCESS: False                  # if first time, should be true
-    DATA_FORMAT: NDCHW
-    DATA_PATH: "/data1/acsp/toolbox_data/scamps/RawData/Train/"           # Raw dataset path, need to be updated
-    CACHED_PATH: "/data1/acsp/finalpreprocess/"     # Processed dataset save path, need to be updated
+    DATA_FORMAT: NCDHW
```
Comment (yahskapar) [inline configs/train_configs/SCAMPS_SCAMPS_MMPD_TSCAN_BASIC.yaml:35]: Change this back to NDCHW

**Review by yahskapar — APPROVED** [review]
Looks good to me. There's still a few places where there appears to be extra single spacing or tab spacing after something like the character string "180" or at the very end of the line. This is very minor, so I'll still go ahead and approve.

Please also be sure to give your code one more quick test by trying out the configs and posting the results here before merging. Thanks!

**Comment by McJackTang** [comment]
PURE_PURE_MMPD_tscan results: 
FFT MAE (FFT Label):2.1197150735294117
FFT RMSE (FFT Label):4.945881743008046
FFT MAPE (FFT Label):2.8992430306314922
FFT Pearson (FFT Label):0.885015925039581

MMPD INFO:
      LIGHT: [1, 2, 3]  # 1 - LED-Low, 2 - LED-high, 3 - Incandescent, 4 - Nature
      MOTION: [1] # 1 - Stationary, 2 - Rotation, 3 - Talking, 4 - Walking
      EXERCISE: [2] # 1 - True, 2 - False
      SKIN_COLOR: [3] # Fitzpatrick Scale Skin Types - 3, 4, 5, 6
      GENDER: [1, 2]  # 1 - Male, 2 - Female
      GLASSER: [1, 2] # 1 - True, 2 - False
      HAIR_COVER: [1, 2] # 1 - True, 2 - False
      MAKEUP: [1, 2] # 1 - True, 2 - False

**Pull request merged**